test/decoders/test_video_decoder_ops.py

```diff
@@ -371,6 +383,68 @@ def test_color_conversion_library_with_down_scaling(
         swscale_frame0, _, _ = get_next_frame(swscale_decoder)
         assert_tensor_equal(filtergraph_frame0, swscale_frame0)
 
+    @pytest.mark.parametrize(
+        "width_scaling_factor,height_scaling_factor",
+        ((1.31, 1.5), (0.71, 0.5), (1.31, 0.7), (0.71, 1.5), (1.0, 1.0)),
+    )
+    @pytest.mark.parametrize("width", [30, 32, 300])
+    @pytest.mark.parametrize("height", [128])
+    def test_color_conversion_library_with_generated_videos(
+        self, tmp_path, width, height, width_scaling_factor, height_scaling_factor
+    ):
+        video_path = f"{tmp_path}/frame_numbers_{width}x{height}.mp4"
+        command = [
+            "ffmpeg",
+            "-y",
+            "-f",
+            "lavfi",
+            "-i",
+            "color=blue",
+            "-pix_fmt",
+            "yuv420p",
+            "-s",
+            f"{width}x{height}",
+            "-c:v",
+            "libx264",
+            "-t",
+            "1",
+            video_path,
+        ]
+        subprocess.check_call(command)
+
+        decoder = create_from_file(str(video_path))
+        scan_all_streams_to_update_metadata(decoder)
+        add_video_stream(decoder)
+        metadata = get_json_metadata(decoder)
+        metadata_dict = json.loads(metadata)
+        assert metadata_dict["width"] == width
+        assert metadata_dict["height"] == height
+
+        target_height = int(height * height_scaling_factor)
+        target_width = int(width * width_scaling_factor)
+        if width_scaling_factor != 1.0:
+            assert target_width != width
+        if height_scaling_factor != 1.0:
+            assert target_height != height
+
+        filtergraph_decoder = create_from_file(str(video_path))
+        _add_video_stream(
+            filtergraph_decoder,
+            width=target_width,
+            height=target_height,
+            color_conversion_library="filtergraph",
+        )
+        filtergraph_frame0, _, _ = get_next_frame(filtergraph_decoder)
+
+        auto_decoder = create_from_file(str(video_path))
```

Comment:
Let's add a comment explaining what we're doing here, because it's kinda subtle.

* We consider filtergraph to the reference color conversion library. Whatever results it produces, we trust.
* We want to use swscale as much as possible, because it's faster. However, it does not work with widths are not a multiple of 32.
* We have logic in the core library to detect when we can safely use swscale based on video dimension.
* In this test, we compare filtergraph to whatever the result of our auto detection is. That means we are _intentionally_ comparing filtergraph to itself sometimes.

Note that I _almost_ made a comment that I was surprised we weren't comparing filtergraph to swscale, and then I realized the above.

We may want to pull this comment all the up to the top of the test function header, since it's really the core thing we're testing.

Reply:
Done.